trading_calendars/exchange_calendar_xbue.py

```diff
@@ -0,0 +1,373 @@
+    offset=pd.DateOffset(weekday=MO(3)),
+)
+
+NationalFlagDayFrom2011 = Holiday(
+    'National Flag Day (2011-present)',
+    month=6,
+    day=20,
+    start_date='2011',
+)
+
+IndependenceDay = Holiday(
+    'Independence Day',
+    month=7,
+    day=9,
+)
+
+SanMartinsDayTo2010 = Holiday(
+    "San Martin's Day",
+    month=8,
+    day=1,
+    end_date='2011',
+    offset=pd.DateOffset(weekday=MO(3)),
+)
+
+SanMartinsDayFrom2012 = Holiday(
+    "San Martin's Day",
+    month=8,
+    day=1,
+    start_date='2012',
+    offset=pd.DateOffset(weekday=MO(3)),
+)
+
+DayOfRespectForCulturalDiversity = Holiday(
+    "Day of Respect for Cultural Diversity",
+    month=10,
+    day=12,
+    observance=cultural_diversity_observance,
+)
+
+BankHoliday = Holiday(
+    'Bank Holiday',
+    month=11,
+    day=6,
+    observance=not_2018,
+)
+
+DayOfNationalSovereigntyTo2016 = Holiday(
+    'Day of National Sovereignty (2010-2016)',
+    month=11,
+    day=1,
+    start_date='2010',
+    end_date='2015',
+    offset=pd.DateOffset(weekday=MO(4)),
+)
+
+DayOfNationalSovereigntyFrom2017 = Holiday(
+    'Day of National Sovereignty (2017-present)',
+    month=11,
+    day=17,
+    start_date='2017',
+    offset=pd.DateOffset(weekday=MO(1)),
+)
+
+ImmaculateConception = immaculate_conception()
+
+ChristmasEve = christmas_eve(
+    days_of_week=WEEKDAYS,
+    observance=not_2018,
+)
+
+ChristmasDay = christmas()
+
+NewYearsEve = Holiday(
+    "New Year's Eve",
+    month=12,
+    day=31,
+    days_of_week=WEEKDAYS,
+    observance=not_2018,
+)
+
+
+class XBUEExchangeCalendar(TradingCalendar):
+    """
+    Exchange calendar for the Buenos Aires Stock Exchange (XBUE).
+
+    Open Time: 11:00 AM, AST
+    Close Time: 5:00 PM, AST
+
+    Regularly-Observed Holidays:
+    - New Year's Day
+    - Carnival Monday (48 days before Easter)
+    - Carnival Tuesday (47 days before Easter)
+    - Truth and Justice Memorial Day (Mar 24)
+    - Malvinas Day (until 2004 first Monday after Mar 31, 2005-present Apr 2)
+    - Good Friday
+    - Labour Day
+    - May Day Revolution (May 25)
+    - Martin Miguel de Guemes Day (Jun 17, 2016-present)
+    - National Flag Day (until 2010 3rd Monday in Jun, 2011-present Jun 20)
+    - Independence Day (Jul 9)
+    - San Martin's Day (3rd Monday in Aug)
+    - Day of Respect for Cultural Diversity (2nd Monday in Oct)
+    - Bank Holiday (Nov 6)
+    - Day of National Sovereignty (4th Monday in Nov)
+    - Immaculate Conception
+    - Christmas Eve (2005-present)
+    - Christmas Day
+    - Last Trading Day of Year (2005-present)
+
+
+    Early Closes:
+    Note - we are unsure if these are early closes or holidays
+    - Christmas Eve
+    - New Year's Eve
+    """
+    name = 'XBUE'
+
+    tz = timezone('America/Argentina/Buenos_Aires')
+
+    open_times = (
+        (None, time(11, 1)),
+    )
+
+    close_times = (
+        (None, time(17, 00)),
+    )
+
+    # TODO: Verify early close time
+    regular_early_close = time(13)
```

Comment:
Is this still a TODO?

Reply:
Yeah it is, Jamie is unsure of the early close time so he said to just make it 1pm and make a note.  Does TODO work here or is there a different way I should be marking that?